packages/sns/src/root.canister.ts

```diff
@@ -0,0 +1,52 @@
+import type { Principal } from "@dfinity/principal";
+import type {
+  CanisterStatusResultV2,
+  _SERVICE as SnsRootCanister,
+} from "../../../candid/sns_root";
+import { idlFactory as certifiedIdlFactory } from "../../../candid/sns_root.certified.idl";
+import { idlFactory } from "../../../candid/sns_root.idl";
+import type { CanisterOptions } from "./types/canister.options";
+import { createServices } from "./utils/actor.utils";
+
+export class RootCanister {
+  private constructor(
+    private readonly service: SnsRootCanister,
+    private readonly certifiedService: SnsRootCanister
+  ) {}
+
+  public static create(options: CanisterOptions<SnsRootCanister>) {
+    const { service, certifiedService } = createServices<SnsRootCanister>({
+      options,
+      idlFactory,
+      certifiedIdlFactory,
+    });
+
+    return new RootCanister(service, certifiedService);
+  }
+
+  // Source code: https://github.com/dfinity/ic/blob/master/rs/sns/root/src/lib.rs
+
+  /**
+   * List and get a summary of the canisters that are part of the Sns.
+   *
+   * Source code: https://github.com/dfinity/ic/blob/master/rs/sns/root/src/lib.rs
+   *
+   * @param {boolean} [certified=true] - Query or update calls
+   * @param {Principal[]} additionalCanisterIds - An optional list of additional canister ids whose status would also have to be requested and added to the returned list
+   *
+   * @returns {Array<[string, Principal, CanisterStatusResultV2]>} - A list of canisters with type ('root' | 'governance' | 'ledger' | 'dapp' | 'sale'), id and status
+   */
+  public canistersSummary = async ({
```

Comment:
yet, are you planning on having a class per canister?

I think that it would be ok that the different canisters is an implementation details.

For example, the devs can do `sns.participateSwap(...)`, not `swapCanister.participate()`

What do you think?

Reply:
We discussed various scenario and decided to stick to classes for separate services with minimal logic and abstraction. Ideally we should be able to write a parser that generate classes and tests from the candid files.

In parallel, for apps such as nns-dapp or others that want to use lots of services, we will provide a wrapper that search and instantiate all canisters of the sns.